experimental/saiexperimentaldashtrustedvni.h

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiexperimentaldashtrustedvni.h
+ *
+ * @brief   This module defines SAI extensions for DASH trusted VNI
+ *
+ * @warning This module is a SAI experimental module
+ */
+
+#if !defined (__SAIEXPERIMENTALDASHTRUSTEDVNI_H_)
+#define __SAIEXPERIMENTALDASHTRUSTEDVNI_H_
+
+#include <saitypesextensions.h>
+
+/**
+ * @defgroup SAIEXPERIMENTALDASHTRUSTEDVNI SAI - Experimental: DASH trusted VNI specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute data for #SAI_TRUSTED_VNI_ENTRY_ATTR_ACTION
+ */
+typedef enum _sai_trusted_vni_entry_action_t
+{
+    SAI_TRUSTED_VNI_ENTRY_ACTION_PERMIT,
+
+} sai_trusted_vni_entry_action_t;
+
+/**
+ * @brief Entry for trusted_vni_entry
+ */
+typedef struct _sai_trusted_vni_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Exact matched key eni_id
+     *
+     * @objects SAI_OBJECT_TYPE_ENI
+     */
+    sai_object_id_t eni_id;
```

Comment:
The HLD allows for a global list of trusted VNIs associated with the appliance instead of a specific VNI (https://github.com/sonic-net/SONiC/blob/master/doc/dash/dash-sonic-hld.md#328-appliance), how do we plan to handle this if eni_id is part of the entry definition? @prsunny

Reply:
Added `sai_global_trusted_vni_entry_t`